tests/simpleKVBC/TesterCRE/main.cpp

```diff
@@ -158,23 +183,9 @@ class KeyExchangeCommandHandler : public IStateHandler {
     return logger_;
   }
   uint16_t clientId_;
-};
-
-class PublicKeyExchangeHandler : public IStateHandler {
- public:
-  bool validate(const State& state) const override {
-    concord::messages::ClientReconfigurationStateReply crep;
-    concord::messages::deserialize(state.data, crep);
-    return std::holds_alternative<concord::messages::ClientExchangePublicKey>(crep.response);
-  }
-  bool execute(const State&, WriteState&) override {
-    LOG_INFO(getLogger(), "restart client components");
-    return true;
-  }
-  logging::Logger getLogger() {
-    static logging::Logger logger_(logging::getLogger("concord.client.reconfiguration.testerCre.PublicKeyExchange"));
-    return logger_;
-  }
+  std::string key_path_;
+  concord::secretsmanager::SecretsManagerPlain sm_;
```

Comment:
I know that's a tester CRE, but can you change it to `std::unique_ptr<ISecretsManagerImpl>` ?

Reply:
fixed